hexa/files/api.py

```diff
@@ -1,161 +1,90 @@
-import base64
-import json
-import os
-import typing
-from dataclasses import dataclass
-from os.path import dirname, isfile, join
+from warnings import warn
 
-import requests
 from django.conf import settings
-from django.core.exceptions import ValidationError
-from google.api_core.exceptions import NotFound
-from google.cloud import storage
-from google.cloud.exceptions import Conflict
-from google.cloud.iam_credentials_v1 import IAMCredentialsClient
-from google.cloud.storage.blob import Blob
-from google.oauth2 import service_account
-from google.protobuf import duration_pb2
 
+from .gcp import GCPClient
+from .s3 import S3Client
 
-@dataclass
-class ObjectsPage:
-    items: typing.List[any]
-    has_next_page: bool
-    has_previous_page: bool
-    page_number: int
+default_mode = settings.WORKSPACE_STORAGE_ENGINE
```

Comment:
Small suggestion: move `settings.WORKSPACE_STORAGE_ENGINE` to `get_storage()`, I think that doing it in the module itself (vs in a function) could be a problem for tests that override Django settings (https://docs.djangoproject.com/en/5.0/topics/testing/tools/#django.test.SimpleTestCase.modify_settings).

If you set he value of `default_mode` outside a function, then the value would be set at import time and you won't be able to override it later.